docs/en/latest/architecture-design/apisix.md

```diff
@@ -38,14 +38,60 @@ The APISIX core handles the important functions like matching Routes, load balan
+
+6.Load Balancing: APISIX provides load balancing capabilities out of the box. It can distribute traffic evenly across multiple backend services, improving reliability and scalability.
+
+7.Caching: APISIX has a built-in caching system that can be used to cache responses from backend services. This can improve performance and reduce the load on the backend services.
+
+8.Service Discovery: APISIX can use service discovery systems, such as Consul or ZooKeeper, to automatically discover backend services and configure routing accordingly.
+
+9.Metrics: APISIX provides metrics and monitoring capabilities that can be used to monitor the performance of the gateway and the backend services.
+
 ## Request handling process
 
 The diagram below shows how APISIX handles an incoming request and applies corresponding Plugins:
 
 ![flow-load-plugin](https://raw.githubusercontent.com/apache/apisix/master/docs/assets/images/flow-load-plugin.png)
+When a request arrives at APISIX, the following steps take place:
+
+APISIX receives the request: When a client sends a request to the API, APISIX receives it and processes it. The request could be an HTTP request or any other protocol that APISIX supports.
+
+Route matching: APISIX checks the incoming request against the configured routes to find a match. Each route is defined by a combination of a URI path, an HTTP method, and any additional criteria, such as headers or query parameters. If a match is found, APISIX proceeds to the next step.
+
+Plugin execution: Once APISIX has identified the correct route, it executes the plugins associated with that route. Plugins can perform a variety of actions, such as authentication, rate limiting, request/response transformation, and more. The plugins are executed in the order specified in the configuration file.
+
+Proxying: After all the plugins have been executed, APISIX proxies the request to the upstream service. The upstream service could be a web application, a microservice, or any other endpoint that can handle the request.
+
+Response processing: When the upstream service returns a response, APISIX processes it according to the plugins associated with the route. Plugins can modify the response, add headers, or perform other actions.
+
+Response delivery: Finally, APISIX delivers the response to the client. If any plugins have modified the response, the modified version is returned to the client.
 
 ## Plugin hierarchy
 
 The chart below shows the order in which different types of Plugin are applied to a request:
 
 ![flow-plugin-internal](https://raw.githubusercontent.com/apache/apisix/master/docs/assets/images/flow-plugin-internal.png)
+
+In APISIX, the order in which plugins are applied to a request is determined by the order in which they are specified in the configuration file. There are several different types of plugins that can be applied to a request, and they can be categorized into three main groups based on when they are executed:
```

Comment:
Information here does not correspond with the digram above and is more geared towards plugin execution order. Plugin execution orders has been covered here: https://apisix.apache.org/docs/apisix/terminology/plugin/#plugins-merging-precedence

Please update.

Reply:
The sentence above `The chart below shows the order in which different types of Plugin are applied to a request:` is actually not accurate for what the diagram is about, so it may have misled you. The diagram is more for the plugin internal.